src/common/pico_base/include/pico.h

```diff
@@ -16,9 +16,18 @@
  * This header may be included by assembly code
 */
 
+#define	__PICO_STRING(x)	#x
+#define	__PICO_XSTRING(x)	__PICO_STRING(x)
+
 #include "pico/types.h"
 #include "pico/version.h"
+
+// PICO_CONFIG: PICO_CONFIG_HEADER, path to header include in place of the default pico/config.h which may be desirable for build systems which can't easily generate the config_autogen header, type=string, default=none, group=pico_base
```

Comment:
When I run the `tools/extract_configs.py` script against this PR, it complains that `string` isn't a valid config type - it currently only expects `int`, `bool` or `enum`.
I _could_ add `string` as an additional config type, but https://github.com/raspberrypi/pico-project-generator/blob/master/pico_project.py#L547 (which is what makes use of the TSV files that  `extract_configs` creates) doesn't support `string` yet either.

Let me know if you want me to add `string` config items to `extract_configs` (would we also want minimum or maximum string-length validation?), and I'll add a corresponding issue to `pico-project-generator`.

Hmmm, although looking at the CONFIG descriptions, I wonder if `filename` would be a more suitable type that just `string`? :thinking:

Reply:
i will just remove the type for now - feel free to add it back later